archive-data/archive.py

```diff
@@ -0,0 +1,204 @@
+
+    return data
+
+
+def convert_to_list(df: str) -> list[tuple]:
+    """Converts a DataFrame to a list of tuples."""
+
+    return df.to_records(index=False).tolist()
+
+
+def load_to_db(conn: connection, data: list[tuple], query: str) -> None:
+    """Loads the data to the database using the query provided."""
+
+    try:
+        with conn.cursor() as cur:
+            cur.executemany(query, data)
+        conn.commit()
+
+    except Error as err:
+
+        conn.rollback()
+        logging.error("Failed to load data: %s", err)
+        if err.pgcode:
+            logging.error("Error code: %s", err.pgcode)
+
+
+def delete_old_data(conn: connection, queries: list[str]):
+    """Deletes old data from 'arrivals' and 'cancellations'
+    tables in the database that are older than 30 days."""
+
+    try:
+
+        with conn.cursor() as cur:
+            cur.execute("BEGIN;")
+            for query in queries:
+                cur.execute(query)
+            conn.commit()
+
+    except Error as err:
+
+        conn.rollback()
+        logging.error("Failed to delete old data: %s", err)
+        if err.pgcode:
+            logging.error("Error code: %s", err.pgcode)
+
+
+def handler(event: dict = None, context: dict = None) -> dict:
+    """
+    Adds logic from main into handler to be used in lambda.
+    """
+
+    station_queries = [S_DELAYS, S_DELAYS_OVER_5_MIN, S_AVG_DELAY, S_TOTAL_ARRIVALS, S_TOTAL_CANCELLATIONS]
+    operator_queries = [O_DELAYS, O_DELAYS_OVER_5_MIN, O_AVG_DELAY, O_TOTAL_ARRIVALS, O_TOTAL_CANCELLATIONS]
+    deletion_queries = [DELETE_OLD_ARRIVAL_DATA, DELETE_OLD_CANCELLATION_DATA]
+
+    load_dotenv()
+    conn = get_db_connection(ENV)
+
+    stations_data = clean_data(get_stations_performance(conn, station_queries))
+    operators_data = clean_data(get_operators_performance(conn, operator_queries))
+
+    load_to_db(conn, convert_to_list(stations_data), INSERT_STATION_PERFORMANCE)
+    load_to_db(conn, convert_to_list(operators_data), INSERT_OPERATOR_PERFORMANCE)
+
+    delete_old_data(conn, deletion_queries)
+
+    close_connection(conn)
+
+    return {
+        "status": "Success!"
+    }
+
+
+if __name__ == "__main__":
+    load_dotenv()
+
+    conn = get_db_connection(ENV)
+
+    station_queries = [S_DELAYS, S_DELAYS_OVER_5_MIN, S_AVG_DELAY, S_TOTAL_ARRIVALS, S_TOTAL_CANCELLATIONS]
+    operator_queries = [O_DELAYS, O_DELAYS_OVER_5_MIN, O_AVG_DELAY, O_TOTAL_ARRIVALS, O_TOTAL_CANCELLATIONS]
+
+    stations_data = clean_data(get_stations_performance(conn, station_queries))
+    operators_data = clean_data(get_operators_performance(conn, operator_queries))
+
+    load_to_db(conn, convert_to_list(stations_data), INSERT_STATION_PERFORMANCE)
+    load_to_db(conn, convert_to_list(operators_data), INSERT_OPERATOR_PERFORMANCE)
+
+    deletion_queries = [DELETE_OLD_ARRIVAL_DATA, DELETE_OLD_CANCELLATION_DATA]
+    delete_old_data(conn, deletion_queries)
+
+    close_connection(conn)
```

Comment:
If this is running as a lambda function, I believe the code under if__name__ == "__main__" won't be executed and so this can be removed.